Doc/whatsnew/3.11.rst

```diff
@@ -425,8 +425,15 @@ runtime. Other optimizations not under this project are listed in `Optimizations
 Faster Startup
 --------------
 
-Static objects
-~~~~~~~~~~~~~~
+Static Code objects
+~~~~~~~~~~~~~~~~~~~
+
+.. sectionauthor:: Kumar Aditya
+
+Python uses code objects to represent executable code for the interpreter
+which is allocated on heap by unmarshalling the bytecode. In Python 3.11,
```

Comment:
This first line is pretty loaded. I don't mean to insult, but as a non-expert in this area, I'm quite lost.

To clarify, do you mean that the previous startup is:
```
marshalled code object (contains bytecode) -----(unmarshall/deserialize)------> heap code object
```
and with the new method, static code objects skip the first part and the unmarshalling phase?

Reply:
Yeah, so here is the entire process:

Python caches the bytecode in `__pycache__` directory to skip the reading the source code  everytime the script is executed, it is converted to a heap allocated code objects which is what the interpreter operates on and these code objects are created by unmarshalling process which reads the bytecode and creates code objects.

So before deepfreeze python execution (assuming bytecode cache exists): 

```console
Read bytecode cache -> unmarshal -> Heap alloc code object -> ceval.c
```

With deepfreeze:
```console
Statically allocated code objects (1) -> ceval.c
```

So deepfreeze skips the first there stages of execution to speed up the execution.

(1) - there is some book keeping to ensure proper deallocation of these code objects and cpython currently does not support immortal objects.

Reply:
@Fidget-Spinner I'll update the docs, I had created this PR just so I don't forget about it. Feel free to ask any thing more about deepfreeze.

Reply:
Ah got it. The bookkeeping about fake "immortal" objects is cool. Didn't know about that.

Reply:
## Without deepfreeze:
```mermaid
flowchart LR;
    id1[Bytecode cache __pycache__] --> id2[Unmarshal];
    id2 --> id3[Heap allocate code object];
    id3 --> id4[Evaluate];
```
----

## With deepfreeze:
```mermaid
flowchart LR;
    id1[Statically allocated code objects] --> id2[Evaluate];
```
----